frontend/src/app/main/data/workspace.cljs

```diff
@@ -2159,6 +2159,24 @@
         (d/dissoc-in state [:workspace-annotations :id-for-create])))))
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Preview blend modes
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn set-preview-blend-mode
+  [ids blend-mode]
+  (ptk/reify ::set-preview-blend-mode
+    ptk/UpdateEvent
+    (update [_ state]
+      (reduce #(assoc-in %1 [:workspace-local :preview-blend-modes %2] blend-mode) state ids))))
```

Comment:
I guess they don't need to be on `:workspace-local`, prefer using a single level... some thing like `:workspace-prevew-blend` as a root key on state